File: cpp/Lec 2 Build-up Logical Thinking (Patterns)/pattern_16.cpp
// A
// BB
// CCC
// DDDD
// EEEEE

// BRUTE FORCE
#include <iostream>
using namespace std;
int main()
{
    int n = 5;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j <= i; j++)
        {
            cout << char('A' + i);
        }
        cout << endl;
    }
    return 0;
}

// BETTER
#include <iostream>
#include <string>
using namespace std;
int main()
{
    int n = 5;
    for (int i = 0; i < n; i++)
    {
        cout << string(i + 1, 'A' + i) << endl;
    }
    return 0;
}

// BEST
#include <iostream>
#include <string>
using namespace std;
int main()
{
    int n = 5;
    string s;
    for (int i = 0; i < n; i++)
    {
        s.assign(i + 1, 'A' + i);
        cout << s << '\n';
    }
    return 0;
}

File: python/Lec 2 Build-up Logical Thinking (Patterns)/pattern_16.py
# A
# BB
# CCC
# DDDD
# EEEEE

# BRUTE FORCE
n = 5
for i in range(n):
    line = ""
    for j in range(i + 1):
        line += chr(ord('A') + i)
    print(line)

# BETTER
n = 5
for i in range(n):
    print(chr(ord('A') + i) * (i + 1))

# BEST
n = 5
print('\n'.join([chr(ord('A') + i) * (i + 1) for i in range(n)]))